airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultNormalizationWorker.java

```diff
@@ -77,6 +88,13 @@ public NormalizationSummary run(final NormalizationInput input, final Path jobRo
         .withStartTime(startTime)
         .withEndTime(endTime);
 
+    if (!traceFailureReasons.isEmpty()) {
+      summary.setFailures(traceFailureReasons);
+      LOGGER.error("Normalization Failed.");
```

Comment:
probably dumb question: is it intentional that this branch doesn't throw an exception?

Reply:
Yes, valid question though because previous to this change normalization would be left to throw and then we'd catch the error as a temporal exception. Instead of relying on throwing (although we can still throw as before), this adds a check on the normalization summary for failure reasons so we can fill the failure reasons in the db and register a fail.